src/progressBar.js

```diff
@@ -0,0 +1,56 @@
+
+export function updateProgressBar() {
+    let tagsMaxCount = 124;
+    let tagsCurrentCount = document.getElementById("list").querySelectorAll("li").length - 1;  // Subtracting 1 because of the "Inputs" list item
+    
+    let progressPercentage = (tagsCurrentCount / tagsMaxCount) * 100;
+
+    let progressBar = document.getElementById('progress-bar');
+    progressBar.value = progressPercentage;
+
+    // FOR SCREEN READERS
+
+    // progress bar label logic
+    let progressDescription = document.getElementById('progress-description')
+    progressDescription.textContent = `Your progress: ${tagsCurrentCount} out of ${tagsMaxCount} tags added.`;
+
+    // MDN DOCS RECOMMEND USING IT, BUT I DID NOT NOTICE ANY DIFFERENCE WHEN SCREEN READER WAS ON.
+    // NVDA only announces the label, while the Narrator announces the label (twice) and the %, but 
+    // this is regardless whether aria-valuetext is used or not.
+    let ariaValueText = `${tagsCurrentCount}`;
+    progressBar.setAttribute('aria-valuetext', ariaValueText);
+
+    // this will announce the current count after each input. TOO MUCH? MAKE IT EVERY 10 INPUTS?
+    // OR NOT AT ALL? JUST UPDATE AT 50% AND 100%?
+    let ariaMessage = `${tagsCurrentCount} added.`;
+    let progressLive = document.getElementById('progress-live');
+    progressLive.textContent = ariaMessage;
+
+    // TEST CASES for updating aria-live region. Note: the values are made up.
```

Comment:
Finally my time for learning all about tests. I wanted to add these as an issue as well 😄